### PR TITLE
feat: enable prometheus metrics for overlaybd

### PR DESCRIPTION
**Pull request by ganeshkumar5699**
**What type of PR is this?**
/kind feature

**What this PR does / why we need it**:
Adding prometheus port for overlaybd metrics. port: 9863 is the default one for overlaybd.

**Which issue(s) this PR fixes**:

Fixes #

**Requirements**:
- [ ] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Special notes for your reviewer**:

**Release note**:
```
none
```

### CODE REVIEW

File: parts/linux/cloud-init/artifacts/cse_config.sh [hunk 1]
```diff
@@ -350,6 +350,7 @@ ensureArtifactStreaming() {
   sudo /opt/acr/tools/overlaybd/enable-http-auth.sh
   sudo /opt/acr/tools/overlaybd/config.sh download.enable false
   sudo /opt/acr/tools/overlaybd/config.sh cacheConfig.cacheSizeGB 32
+  sudo /opt/acr/tools/overlaybd/config.sh exporterConfig.port 9863
```
Comment (jason1028kr) [inline parts/linux/cloud-init/artifacts/cse_config.sh:353]: Should `exporterConfig.enable` also be set to true (https://github.com/containerd/overlaybd#:~:text=exporterConfig.enable,show%20Prometheus%20metrics.)? I am not sure how `acr/.../overlaybd/config.sh` looks, so just checking.

Comment (ganeshkumar5699) [reply parts/linux/cloud-init/artifacts/cse_config.sh:353]: Good point, yes, that's needed. Thanks! config.sh just sets the config for overlaybd. Will update it.

File: parts/linux/cloud-init/artifacts/cse_config.sh [hunk 2]
```diff
@@ -350,6 +350,8 @@ ensureArtifactStreaming() {
   sudo /opt/acr/tools/overlaybd/enable-http-auth.sh
   sudo /opt/acr/tools/overlaybd/config.sh download.enable false
   sudo /opt/acr/tools/overlaybd/config.sh cacheConfig.cacheSizeGB 32
+  sudo /opt/acr/tools/overlaybd/config.sh exporterConfig.enable true
+  sudo /opt/acr/tools/overlaybd/config.sh exporterConfig.port 9863
```
[the remark this answers is not in the document]
Comment (juan-lee) [reply parts/linux/cloud-init/artifacts/cse_config.sh:354]: seems like no. Let's create a work item to track this for GA.

**Review by juan-lee — APPROVED** [review]

File: parts/linux/cloud-init/artifacts/cse_config.sh (hunk 2 above)
[the remark this answers is not in the document]
Comment (ganeshkumar5699) [reply parts/linux/cloud-init/artifacts/cse_config.sh:354]: TLS is not supported -- filed work item for GA.

**Review by cameronmeissner — APPROVED** [review]

**Pull request merged**